crates/wasmtime/src/config.rs

```diff
@@ -29,7 +29,8 @@ pub struct Config {
     #[cfg(feature = "cache")]
     pub(crate) cache_config: CacheConfig,
     pub(crate) profiler: Arc<dyn ProfilingAgent>,
-    pub(crate) memory_creator: Option<MemoryCreatorProxy>,
+    pub(crate) instance_allocator: Option<Arc<dyn InstanceAllocator>>,
+    pub(crate) default_instance_allocator: DefaultInstanceAllocator,
```

Comment:
Why do we have both of these? I would have expected just `instance_allocator` (without the `Option`) which is initialized by default to `Arc::new(DefaultInstanceAllocator)`.

Reply:
So the instance allocator is really meant for allocating *module* instances and not for the various "instantiations" Wasmtime does internally for representing host tables, memories, globals, and functions.

The "default" allocator is used exclusively for these internal instantiations as well as for module instantiations when a user does not provide an alternative instance allocator.

I can improve the comments around this to make that distinction clearer, but I'm also open to an alternative implementation.

Reply:
This field has now been removed.

Where instance handles are created for host objects, the default instance allocator is explicitly used with a comment.